scripts/o3de/o3de/export_project.py

```diff
@@ -136,3 +207,322 @@ def add_args(subparsers) -> None:
 
+
+def get_asset_processor_batch_path(non_mono_build_path: pathlib.Path):
+    return non_mono_build_path / f'bin/profile/AssetProcessorBatch{EXECUTABLE_EXTENSION}'
+
+
+def get_asset_bundler_batch_path(non_mono_build_path: pathlib.Path):
+    return non_mono_build_path / f'bin/profile/AssetBundlerBatch{EXECUTABLE_EXTENSION}'
+
+
+def build_assets(ctx: O3DEScriptExportContext,
+                 tools_build_path: pathlib.Path,
+                 fail_on_ap_errors: bool,
+                 logger: logging.Logger = None) -> None:
+    """
+    Build the assets for the project
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to locate AssetProcessorBatch
+    @param fail_on_ap_errors:   Option to fail the whole process if an error occurs during asset processing
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Make sure `AssetProcessorBatch` is available
+    asset_processor_batch_path = get_asset_processor_batch_path(tools_build_path)
+    if not asset_processor_batch_path.exists():
+        raise ExportProjectError("Missing AssetProcessorBatch. The pre-requisite tools must be built first.")
+
+    # Build the project assets with the {project_name}.Assets custom target
+    if logger:
+        logger.info(f"Processing assets for {ctx.project_name}")
+
+    cmake_build_assets_command = [asset_processor_batch_path, "--project-path", ctx.project_path]
+    ret = process_command(cmake_build_assets_command,
+                          cwd=ctx.engine_path if ctx.is_engine_centric else ctx.project_path)
+    if ret != 0:
+        if fail_on_ap_errors:
+            raise ExportProjectError(f"Error building assets for project {ctx.project_name}.")
+        else:
+            if logger:
+                logger.warning("Some assets failed to processed.")
+    if logger:
+        logger.info(f"Completed processing assets for {ctx.project_name}!")
+
+
+def build_export_toolchain(ctx: O3DEScriptExportContext,
+                           tools_build_path: pathlib.Path,
+                           logger: logging.Logger = None) -> None:
+    """
+    Build (or rebuild) the export tool chain (AssetProcessorBatch and AssetBundlerBatch)
+
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to create the build project for the tools
+    @param build_cwd:           The working directory to use when executing A
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Generate the project for export toolchain
+    cmake_configure_command = ["cmake", "-B", tools_build_path]
+
+    if ctx.is_engine_centric:
+        cmake_configure_command.extend(["-S", ctx.engine_path])
+    else:
+        cmake_configure_command.extend(["-S", ctx.project_path])
+
+    # Generate the project for the pre-requisite tools
+    if GENERATOR:
+        cmake_configure_command.extend(["-G", GENERATOR])
```

Comment:
If Generator is always auto-assigned at the start of `export_project.py`, then is there anyway to override the preferred generator if needed?

Reply:
I think we should keep the options as light as possible and not introduce any options, especially since the generator doesn't provide us any benefits for this process other than Ninja supporting parallel builds (Unix doesnt), but that will be addressed as mentioned above

Reply:
that's fair, I think we will only introduce it if there's a real need for it